dll/win32/shell32/CDefView.cpp

```diff
@@ -1196,6 +1258,19 @@ LRESULT CDefView::OnCreate(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL &bHandl
 
     m_hAccel = LoadAcceleratorsW(shell32_hInstance, MAKEINTRESOURCEW(IDA_SHELLVIEW));
 
+    BOOL bPreviousParentSpecial = m_isParentFolderSpecial;
+
+    // A folder is special if it is the Desktop folder, a network folder, or a Control Panel folder
+    m_isParentFolderSpecial = _ILIsDesktop(m_pidlParent) || _ILIsNetHood(m_pidlParent) 
+        || _ILIsControlPanel(ILFindLastID(m_pidlParent));
+
+    // Only force Status Bar part refresh if the state changed from the preivous folder
+    if (bPreviousParentSpecial != m_isParentFolderSpecial)
+    {
+        // This handles changing StatusBar parts
+        _ForceStatusBarResize();
```

Comment:
Can you explain a bit more why this is needed?

Reply:
There are folders where it shows 3 status bar parts or one. If a user navigates to a folder where it changes the number status bar parts we want to update the UI.

Reply:
Please add a comment here that we need to resize the status bar because special folders have one part and the others have one.